### DISCUSSION TITLE
How many files will there be in each folder in the repository?

### DISCUSSION THREAD

**Discussion by gdtv** [rustic-rs/rustic#1180] [Q&A] [answered]
How many files will there be in each folder in the repository?
Can the max number of files in each folder be config? Because some storage providers limit the number of files per folder.

**Comment by kapitainsky** [comment, marked as answer]
It is the same answer (as the repo format is the same) as for the same question you asked on restic forum...

https://forum.restic.net/t/how-many-files-will-there-be-in-each-folder-in-the-repository/8022/2?u=kapitainsky

```
number of files per dir ≈ repo size / 256 / pack size
```

You can control pack size if needed.

**Reply by gdtv** [reply]
Thank you for your reply. 
Yes, I asked the same question to restic. I want to know if there is a different way to control the number of files in a folder in Rustic.
In fact, I have been using Duplicati to back up files before, but the number of files in the same folder level is too large, which exceeds the limit of the cloud storage provider. I have to consider using restic or rustic instead.

**Comment by aawsome** [comment]
@gdtv About pack files, see the answer of @kapitainsky. Note that rustic offer much more possibilities to control pack size (see `rustic config --help`) and has per default higher pack sizes than restic.

Besides pack files, there are also  snapshot files and index files.
- About snapshot files: Those are simply the number of snapshots which are all saved in a single folder. You can reduce that number by regularly running `rustic forget` using a suitable policy. Note that if you run `backup` very often and don't run `forget`, you can get quickly thousands of snapshots.
- About index files: Those are also saved in a single folder, The number depends  mainly on the repository size. Typically it should be much lower than number of pack files per pack file folder. But note that it is possible to have lots of index files if you often run `backup`. Running `rustic prune` will however repack index files which reduces the number a lot. So, regularly run `prune`!
